.github/workflows/comment.yaml

```diff
@@ -96,7 +85,9 @@ jobs:
                     issue_number: issue_number,
                 });
                 const githubActionsComments = existingComments.data.filter(
-                    comment => comment.user.login == 'github-actions[bot]'
+                    comment =>
+                      comment.user.login == 'github-actions[bot]' &&
+                      !comment.body.startsWith(header),
```

Comment:
here is that filtering.   We want to delete comments if they do not apply to this commit.

Reply:
I think the problem here is that for multiple results, as they are all running on the same commit and thus share the same commit SHA. As a result, multiple `comment-on-prs` jobs will overwrite each other, leaving only one comment in the pull request thread.

I would try adding the name of the result to the header so they won't overlap with each other.

Reply:
I agree @weiiwang01 .  If there are multiple reports on a PR, we want to delete comments if they originated from a previous commit id. I know that since this is running in a matrix, we want to skip deleting a comment if it originates from the same sha256, hence the `!comment.body.startsWith(header)`

I envision that if there are 3 reports, there will be 3 * N comments showing up in the report -- all with the same header associated with that commit id (N being the list of comments in the report).

Reply:
Yes, I think you are right.